example_project_bzl_4/WORKSPACE

```diff
@@ -0,0 +1,45 @@
+workspace(name = "example_project_bzl_4")
+
+load(
+    "@bazel_tools//tools/build_defs/repo:http.bzl",
+    "http_archive",
+)
+
+http_archive(
+    name = "rules_cc",
+    urls = ["https://github.com/bazelbuild/rules_cc/releases/download/0.0.1/rules_cc-0.0.1.tar.gz"],
+    sha256 = "4dccbfd22c0def164c8f47458bd50e0c7148f3d92002cdb459c2a96a68498241",
+)
+
+http_archive(
+    name = "io_tweag_rules_nixpkgs",
+    sha256 = "7efb0f82cda5cdaa8e96b687734c5704b91353561d7889e5d84a13b6d30f4cb8",
+    strip_prefix = "rules_nixpkgs-05a445575de51872f528c537cc7ff3ab11114f21",
+    urls = ["https://github.com/tweag/rules_nixpkgs/archive/05a445575de51872f528c537cc7ff3ab11114f21.tar.gz"],
+)
+
+load("@io_tweag_rules_nixpkgs//nixpkgs:repositories.bzl", "rules_nixpkgs_dependencies")
+rules_nixpkgs_dependencies()
+
+load("@io_tweag_rules_nixpkgs//nixpkgs:nixpkgs.bzl", "nixpkgs_git_repository", "nixpkgs_package", "nixpkgs_cc_configure")
+
+nixpkgs_git_repository(
+    name = "nixpkgs",
+    revision = "21.11",
+    sha256 = "c77bb41cf5dd82f4718fa789d49363f512bb6fa6bc25f8d60902fe2d698ed7cc",
+)
+
+nixpkgs_cc_configure(
+    name = "toolchain-linux-x86_64",
+    repositories = {
+        "nixpkgs": "@nixpkgs",
+    },
+    exec_constraints = [
+        "@platforms//cpu:x86_64",
+        "@platforms//os:linux",
```

Comment:
We could add in support, at least theoretical one for OSX, if that's a hot topic.
I would however vote to introduce OSX changes in another PR, not to bloat this one with random R&D.

Reply:
Ditto.